### PR TITLE
Add chat schema example for mlflow

### PR DESCRIPTION
**Pull request by dakinggg**
Screenshot 2024-03-22 at 5 53 19 PM
Screenshot 2024-03-22 at 5 53 23 PM

### CODE REVIEW

File: llmfoundry/callbacks/hf_checkpointer.py [hunk 1]
```diff
@@ -129,39 +129,30 @@ def __init__(
...
-                ColSpec('double', 'temperature', optional=True),
-                ColSpec('integer', 'max_tokens', optional=True),
-                ColSpec('string', 'stop', optional=True),
-                ColSpec('integer', 'candidate_count', optional=True)
-            ])
-
-            output_schema = Schema([ColSpec('string', 'predictions')])
-
-            default_signature = ModelSignature(inputs=input_schema,
-                                               outputs=output_schema)
```
Comment (irenedea) [inline llmfoundry/callbacks/hf_checkpointer.py:157]: why don't we need this anymore?

**Review by irenedea — APPROVED** [review]

File: llmfoundry/callbacks/hf_checkpointer.py [hunk 2]
```diff
@@ -129,39 +129,30 @@ def __init__(
...
+            mlflow_logging_config['metadata'] = passed_metadata
+            mlflow_logging_config.setdefault('task', 'llm/v1/completions')
 
             default_input_example = {
                 'prompt': np.array(['What is Machine Learning?'])
             }
+            is_chat = mlflow_logging_config['task'].endswith(
```
Comment (nancyhung) [inline llmfoundry/callbacks/hf_checkpointer.py:142]:
```suggestion
            is_chat = mlflow_logging_config['task'].contains(
```

**Review by nancyhung — APPROVED** [review]
lgtm

File: llmfoundry/callbacks/hf_checkpointer.py (hunk 2 above)
Comment (nancyhung) [reply llmfoundry/callbacks/hf_checkpointer.py:142]: actually probably fine to do this

File: llmfoundry/callbacks/hf_checkpointer.py (hunk 1 above)
Comment (dakinggg) [reply llmfoundry/callbacks/hf_checkpointer.py:157]: they told me that when using `llm/v1/{completions|chat}`, the input schema and signature would be added automatically, so no need to hardcode here, which would be more brittle.

**Pull request merged**